src/Symfony/Component/Mime/Part/TextPart.php

```diff
@@ -63,9 +63,6 @@ public function __construct($body, ?string $charset = 'utf-8', string $subtype =
         if (null === $encoding) {
             $this->encoding = $this->chooseEncoding();
         } else {
-            if ('quoted-printable' !== $encoding && 'base64' !== $encoding && '8bit' !== $encoding) {
-                throw new InvalidArgumentException(sprintf('The encoding must be one of "quoted-printable", "base64", or "8bit" ("%s" given).', $encoding));
-            }
```

Comment:
This basically means that constructing a `TextPart` or `DataPart` with an invalid encoding won't fail anymore. That error is now deferred to when we render the part. That's unfortunate.

Reply:
Yeah, I was thinking about that as well, but since you can add them at runtime now there's not much choice.
I could modify it in a way that you pass the extra encodes in the constructor and then we could move the check back up there again.
This again would clash with the review from above about it not being clear that it's static though.

Reply:
But if the mutator were static, I could add a new encoder before calling the constructor. And then, the constructor could perform its check again, right?

Reply:
Yeah, I also realised that while I was working on it and rewrote it that way with the last commit